src/ess/livedata/dashboard/config_store.py

```diff
@@ -68,3 +99,280 @@ def _evict_oldest(self) -> None:
+
+    def __setitem__(self, key: WorkflowId, value: dict[str, Any]) -> None:
+        """Save configuration with automatic LRU eviction and file persistence."""
+        with self._lock:
+            super().__setitem__(key, value)
+
+            # Automatic LRU eviction if limit exceeded
+            if self._max_configs and len(self.data) > self._max_configs:
+                self._evict_oldest()
+
+            # Persist to file after modification
+            self._save_to_file()
+
+    def __delitem__(self, key: WorkflowId) -> None:
+        """Delete configuration and persist to file."""
+        with self._lock:
+            super().__delitem__(key)
+            self._save_to_file()
+
+    def _evict_oldest(self) -> None:
+        """Remove oldest configs based on cleanup fraction."""
+        num_to_remove = max(1, int(len(self.data) * self._cleanup_fraction))
+        # Dict maintains insertion order in Python 3.7+
+        oldest_keys = list(self.data.keys())[:num_to_remove]
+        for key in oldest_keys:
+            # Use dict.pop to avoid triggering __delitem__ which would save repeatedly
+            self.data.pop(key, None)
+
+    def _load_from_file(self) -> None:
+        """Load configurations from YAML file."""
+        if not self._file_path.exists():
+            logger.debug(
+                "Config file %s does not exist, starting empty", self._file_path
+            )
+            return
+
+        try:
+            with open(self._file_path) as f:
+                raw_data = yaml.safe_load(f)
+                if raw_data is None:
+                    logger.debug("Config file %s is empty", self._file_path)
+                    return
+
+                # Deserialize WorkflowId keys from strings
+                for key_str, value in raw_data.items():
+                    try:
+                        workflow_id = WorkflowId.from_string(key_str)
+                        self.data[workflow_id] = value
+                    except (ValueError, KeyError) as e:
+                        logger.warning(
+                            "Skipping invalid config entry '%s': %s", key_str, e
+                        )
+
+        except yaml.YAMLError as e:
+            logger.error("Failed to parse config file %s: %s", self._file_path, e)
+            logger.warning("Starting with empty config store")
+        except Exception as e:
+            logger.error(
+                "Unexpected error loading config file %s: %s", self._file_path, e
+            )
+            logger.warning("Starting with empty config store")
+
+    def _save_to_file(self) -> None:
+        """Save configurations to YAML file using atomic write operation."""
+        try:
+            # Serialize WorkflowId keys to strings for YAML
+            serialized_data = {str(key): value for key, value in self.data.items()}
+
+            # Write atomically: write to temp file, then rename
+            temp_path = self._file_path.with_suffix('.tmp')
```

Comment:
This uses the same temp file for a store across threads. So make sure that the lock is held when calling this function! (It is currently, but code changes.) I don't know if `threading.Lock` has a way to prove that you are holding it. So unless you want to build such a system, you may have to just be very careful.